xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp

```diff
@@ -114,8 +116,21 @@ bool CRendererDXVA::Configure(const VideoPicture& picture, float fps, unsigned o
       if (CServiceBroker::GetLogging().IsLogLevelLogged(LOGDEBUG))
         m_processor->ListSupportedConversions(dxgi_format, dest_format, picture);
 
-      if (m_processor->IsFormatConversionSupported(dxgi_format, dest_format, picture))
+      if (!m_processor->IsFormatConversionSupported(dxgi_format, dest_format, picture))
+        CLog::LogF(LOGWARNING, "DXGI color format conversion may be not full supported.");
```

Comment:
what this mean? if format isn't supported dxva will not work and another renderer must be selected.

Reply:
Really this is not related to this PR and reverted for now but here we have various problems:

`IsFormatConversionSupported` is flaky test and fails with "false positives" while video conversion works and picture is correct.

This is not right place for fallback to another render method because returning false here hangs execution. This should be done much before (as already is done) here:

https://github.com/xbmc/xbmc/blob/31124ad11bd289a115851244ac953b8a181b0a89/xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp#L59-L68

Currently fails for HDR10 content when check of BT.2020 color space is already implemented in https://github.com/xbmc/xbmc/pull/23109 and works fine.

See: https://github.com/xbmc/xbmc/pull/23109#issuecomment-1501163850

In my opinion this check is redundant and its usefulness is limited to helping find/debug issues but should be removed in v21 final.

Reply:
my 2 cents, I added this bit originally and know something has to be done before release... other PR, other discussion to have.

I wouldn't call it flaky. When it returns false, you can still sometimes get an output, but in my experience it's not exactly what was asked for.
For example, that function says that my AMD doesn't support any processor RGB limited range output, only full range.
If I bypass that warning and still request a conversion to limited range, I still get a picture, but it's full range instead of the requested limited.

So at least in the case of AMD, it looks like the driver is sometimes able to fall back to one of its supported conversions.

It doesn't won't work for everything. When there is no fallback you get black screen + the dxva blit returns an error.

Reply:
> other PR, other discussion to have.

It's not my fault if every time I try Kodi with a new HW it crashes here😛 

Initially worked fine with RTX 2060 but failed in Intel NUC.... fixed but now upgraded to RTX 4070 and still working fine in NUC but fails in NVIDIA ..... is only 4070? is new Nvidia driver? Investigating this is very tedious and honestly I don't feel like it. I can lose many hours. 😒

But the main argument is not that the method fails, but that even if it works fine, it is not useful to stop execution here. It's like throwing an exception to exit. The ideal would be to find out why it fails (I will surely do this at some point) but when it works well move that check to `CRendererDXVA::GetWeight` and fallback other render method.

Anyway this is of topic for this PR.